lib/compress/huf_compress.c

```diff
@@ -693,7 +701,7 @@ HUF_compress_internal (void* dst, size_t dstSize,
     huffLog = HUF_optimalTableLog(huffLog, srcSize, maxSymbolValue);
     {   size_t const maxBits = HUF_buildCTable_wksp(table->CTable, table->count,
                                             maxSymbolValue, huffLog,
-                                            table->nodeTable, sizeof(table->nodeTable));
+                                            table->nodeTable, ((BYTE*)workSpace) + wkspSize - (BYTE*)table->nodeTable);
```

Comment:
This arithmetic feels weird. What's the goal ?

Reply:
I believe you need to update the definition of `HUF_compress_tables_t`,
so that the space previously reserved as `huffNodeTable nodeTable;` now becomes larger, as `HUF_buildCTable_wksp_tables buildCTable_wksp;` instead.
And then you just pass this section as the `workspace` of `HUF_buildCTable_wksp()`.